core/src/main/java/com/zextras/carbonio/files/dal/repositories/interfaces/UserRepository.java

```diff
@@ -5,22 +5,22 @@
 package com.zextras.carbonio.files.dal.repositories.interfaces;
 
 import com.zextras.carbonio.files.dal.dao.User;
+import com.zextras.carbonio.files.dal.dao.UserMyself;
 import com.zextras.carbonio.usermanagement.entities.UserId;
 import io.vavr.control.Try;
+
 import java.util.Optional;
 
 public interface UserRepository {
 
-  Optional<User> getUserById(
-    String cookies,
-    String userId
-  );
+  /**
+   * no cache on this one since it is not requested often and we always want the updated version
+   */
+  Optional<UserMyself> getUserMyselfByCookieNotCached(String cookies);
```

Comment:
```suggestion
  /**
   * Retrieves a {@link UserMyself} without caching the response and/or checking if it already retrieved since it is not requested often and we always want the updated version.
   *
   * @param cookies is a {@link String} representing the cookie of the requester used to fetch all the metadata of the requester itself
   * @return a {@link Optional} of the requested {@link UserMyself} if the cookie is valid, otherwise it returns an {@link Optional#empty}
   */
  Optional<UserMyself> getUserMyselfByCookieNotCached(String cookies);
```